modules/javafx.controls/src/test/java/test/javafx/scene/control/SplitPaneTest.java

```diff
@@ -1329,4 +1335,55 @@ private double convertDividerPostionToAbsolutePostion(double pos, double edge) {
 
         sl.dispose();
     }
+
+    /**
+     * Verifies that a divider position change of the {@link SplitPane} does not hang the layout.
+     * Previously, this may happen when the divider position changed to a large number (>1),
+     * which can hang the layout as it resulted in multiple layout requests (through SplitPaneSkin.layoutChildren).
+     */
+    @Test
+    public void testDividerOverOneDoesNotHangLayout() {
+        testSetDividerPositionDoesNotHangLayout(10);
+    }
+
+    /**
+     * Verifies that a divider position change of the {@link SplitPane} does not hang the layout.
+     * Previously, this may happen when the divider position changed to a negative number (<1),
+     * which can hang the layout as it resulted in multiple layout requests (through SplitPaneSkin.layoutChildren).
+     */
+    @Test
+    public void testDividerUnderZeroDoesNotHangLayout() {
+        testSetDividerPositionDoesNotHangLayout(-1);
+    }
+
+    private void testSetDividerPositionDoesNotHangLayout(double dividerPosition) {
+        AtomicInteger layoutCounter = new AtomicInteger();
+        ComboBox<String> cbx = new ComboBox<>(FXCollections.observableArrayList("1", "2", "3")) {
+            @Override
+            protected void layoutChildren() {
+                layoutCounter.incrementAndGet();
+                super.layoutChildren();
+            }
+        };
+        SplitPane pane = new SplitPane(new Label("AAAAA"), new TabPane(new Tab("Test", cbx)));
+        StackPane root = new StackPane(pane);
+
+        StageLoader stageLoader = new StageLoader(root);
+
+        Toolkit.getToolkit().firePulse();
+
+        pane.setDividerPosition(0, dividerPosition);
+
+        Toolkit.getToolkit().firePulse();
+
+        // Reset layout counter
+        layoutCounter.set(0);
+
+        cbx.getSelectionModel().select(0);
+        Toolkit.getToolkit().firePulse();
+
+        assertTrue(layoutCounter.get() > 0);
+        stageLoader.dispose();
+    }
```

Comment:
the stageLoader is not disposed if the test fails - a (recently introduced :) general pattern is to use an instance field that's disposed in the test cleanup

Reply:
Hm, is this really needed? Not sure, are there any side effects by the `StageLoader` like this when a test failed?
Just asking since the `StageLoader` is used a lot like this.
And since the tests normally run green unless you may change something locally (on the JavaFX Pipeline it should never be red), it would probably never affect anything (or maybe it does?).

Reply:
> Hm, is this really needed? 

yes, IMO, we want the exact same cleanup for passing/failing tests. So either dispose is required always (then need to make sure it's called on failure) or not required always (then all its calls would be noise).

> Not sure, are there any side effects by the `StageLoader` like this when a test failed?  Just asking since the `StageLoader` is used a lot like this. 

don't now (and doesn't matter, what matters is the guaranteed cleanup) - and aware of those slightly fishy patterns, we all learn :) Faintly remember having discussed the point in a PR (can't find it right now, though), and just as faintly remember the outcome was to guarantee the cleanup in new tests.

Reply:
ah okay. Was just confusing for me since I never read that and I think some recent PRs still had this pattern, e.g. also the touch table scrolling PR I had a look at yesterday.

Maybe for future it makes sense to have an abstract test class with the stage loader setup already in place.

Reply:
just for reference - found the source of my [faintly remember](https://github.com/openjdk/jfx/pull/337/files/8a6fc1cf6cad2c453de17b71777ddd63fadb539e#r510975640)

Reply:
Done. Also adjusted the copyright year ;)